terraform/builder-monolith.tf

```diff
@@ -20,6 +20,21 @@ resource "aws_instance" "monolith" {
         agent       = "${var.connection_agent}"
     }
 
+    ebs_block_device {
```

Comment:
I spent most of today playing with this. I think having the ebs volume as part of the instance lifecycle is the right thing, because if we're treating any volume created outside the instance as an attachment with it, then we're basically doing that anyway, but we run into provisioning timing issues and workarounds. Specifically, [this issue](https://github.com/hashicorp/terraform/issues/6365) is precisely what we have to deal with, which requires something like this:

```terraform
resource "template_file" "monolith_userdata" {
    template = "${file("${path.module}/scripts/habfs.sh")}"
}

resource "aws_instance" "monolith" {
    // ...
    user_data     = "${template_file.monolith_userdata.rendered}"
    // ... 
}
```

Where `habfs.sh` looks like this:

```sh
#!/bin/bash

while [ `lsblk -n | grep -c 'xvdf'` -ne 1 ]
do
  echo "Waiting for /dev/xvdf to become available"
  sleep 10
done

mkfs.ext4 /dev/xvdf
mount /dev/xvdf /mnt
echo '/dev/xvdf /hab ext4 defaults 0 0' | tee -a /etc/fstab
mkdir -p /mnt/hab
ln -s /mnt/hab /hab
```

But then we have to handle getting the initial hab install in the right place as the above happens after the instance is provisioned.

Overall I think having the EBS volume with the lifecycle of the instance the right thing, and if it gets destroyed in acceptance or other non-production/live environments, it's probably not a big deal.

For production/live, we can create volumes out of a snapshot. We should set up some kind of snapshot backup schedule. I created a snapshot earlier for testing, and in case something happens we have it available at least.